src/librustdoc/clean/types.rs

```diff
@@ -641,6 +643,53 @@ impl Item {
             _ => false,
         }
     }
+
+    crate fn hir_id(&self) -> hir::HirId {
+        hir::HirId::make_owner(
+            self.def_id.as_def_id().unwrap().as_local().expect("not local DefId"),
+        )
+    }
+
+    crate fn header(&self, tcx: TyCtxt<'_>) -> Option<hir::FnHeader> {
+        let header = match *self.kind {
+            ItemKind::ForeignFunctionItem(_) => {
+                let abi = tcx.hir().get_foreign_abi(self.hir_id());
```

Comment:
How can you be sure this is a local item?

Reply:
`ForeignFunctionItem` is created in one place where we always called `self.hir_id`. So I think it's safe to assume that `ForeignFunctionItem` is always a local item, as weird as it may seem...

Reply:
After investigating a bit more, it seems like it's not a `ForeignFunctionItem` anymore when it's imported from another crate but simply a `FunctionItem`. I'll add a test to be sure it's really the case though.

Reply:
We already have tests checking that actually. So if they pass, I guess it's fine?

Reply:
On how I checked, I took the test from `src/test/rustdoc/index-page.rs` and reexported the extern `foo_ffn` function, which gave:

```
Item { name: Some("foo_ffn"), attrs: Attributes { doc_strings: [], other_attrs: [] }, visibility: Public, kind: FunctionItem(Function { decl: FnDecl { inputs: Arguments { values: [] }, output: DefaultReturn, c_variadic: false }, generics: Generics { params: [], where_predicates: [] } }), def_id: DefId(DefId(20:5 ~ all_item_types[6512]::{extern#0}::foo_ffn)), cfg: None }
```